webpack/ForemanTasks/Components/TasksTable/TasksTableSchema.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import {
+  column,
+  sortableColumn,
+  headerFormatterWithProps,
+  cellFormatter,
+} from 'foremanReact/components/common/table';
+import { translate as __ } from 'foremanReact/common/I18n';
+import LongDateTime from 'foremanReact/components/common/dates/LongDateTime';
+
+const headFormat = [headerFormatterWithProps];
+const cellFormat = [cellFormatter];
+
+const sortControllerFactory = (setSort, sortBy, sortOrder) => ({
+  apply: (by, order) => {
+    setSort({ by, order });
+  },
+  property: sortBy,
+  order: sortOrder,
+});
+
+const actionCellFormatter = url => (value, { rowData: { id } }) =>
+  cellFormatter(<a href={`/${url}/${id}`}>{value}</a>);
+
+const dateCellFormmatter = value => (
+  <LongDateTime seconds date={value} defaultValue={__('N/A')} />
+);
+
+/**
+ * Generate a table schema to the Hardware Tasks page.
+ * @param  {Function} setSort a Redux async action that sets new sort values
+ * @param  {String}   by      by which column the table is sorted.
+ *                            If none then set it to undefined/null.
+ * @param  {String}   order   in what order to sort a column. If none then set it to undefined/null.
+ *                            Otherwise, 'ASC' for ascending and 'DESC' for descending
+ * @return {Array}
+ */
+const createTasksTableSchema = (setSort, by, order) => {
+  const sortController = sortControllerFactory(setSort, by, order);
```

Comment:
Does the `sortControllerFactory` actually need now? e.g.

```js
const sortController = {
  property: sortBy,
  order: sortOrder,
  apply: (by, order) => setSort({ by, order }),
};
```

Reply:
Nope, removed, thanks